src/filesys.cpp

```diff
@@ -740,5 +744,23 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c)
+{
+	return !(isalnum(c) || c == '-' || c == '_' || c == ' ');
+}
+
+std::string SanitizeFilename(std::string input)
+{
+	input.erase(remove_if(input.begin(), input.end(), bad_char), input.end());
+	input.erase(input.find_last_not_of(" ") + 1);
+	for (unsigned int a = 0; a < sizeof(bad_names) / sizeof(bad_names[0]); a++) {
```

Comment:
i think we have a macro for the `sizeof(bad_names) / sizeof(bad_names[0])` thing
also just use `int`